lib/std/os/windows.zig

```diff
@@ -2871,9 +2874,137 @@ pub const PROV_RSA_FULL = 1;
 
 pub const REGSAM = ACCESS_MASK;
 pub const ACCESS_MASK = DWORD;
-pub const HKEY = *opaque {};
 pub const LSTATUS = LONG;
 
+pub const HKEY = HANDLE;
```

Comment:
It seems like this change was made to be able to use `HKEY` with `RegOpenKeyExW` / `NtOpenKey` more easily, but the current code doesn't use either of those functions AFAICT so it might be worth backing out this change (and also removing the `RegOpenKeyExW` / `NtOpenKey` definitions).

Reply:
Oh right, yeah, will do! I would like to point out though that internally `HKEY` and `HANDLE` seem to be one and the same minus a linear difference in masked out pointer bits. I actually tried implementing `RegOpenKeyExW` difectly in Zig using only ntdll primitives but sadly got stuck at some internal static key stores such as a static preloaded map of system hive `HKEY`s such as `HKEY_LOCAL_MACHINE` to a registry `HANDLE`.

Reply:
Reverted in d19f74a

Reply:
> Oh right, yeah, will do! I would like to point out though that internally `HKEY` and `HANDLE` seem to be one and the same minus a linear difference in masked out pointer bits.

Yeah, when `STRICT` is not defined that's literally true--all `HANDLE` types are just `void*`. From https://learn.microsoft.com/en-us/windows/win32/winprog/enabling-strict:

> Without STRICT, all handles are defined as HANDLE, so the compiler does not prevent you from using one type of handle where another type is expected

From my local `WinDef.h` & `WinNT.h`:

```c
// WinDef.h
DECLARE_HANDLE(HKEY);
DECLARE_HANDLE(HBRUSH);
etc ...

// WinNT.h
#ifdef STRICT
typedef void *HANDLE;
#define DECLARE_HANDLE(name) struct name##__ { int unused; }; typedef struct name##__ *name
#else
typedef PVOID HANDLE;
#define DECLARE_HANDLE(name) typedef HANDLE name
#endif
```

Using `*opaque {}` for each `HANDLE` type gives Zig the benefits of `STRICT` with the tradeoff that it's harder to convert between `HANDLE` types, but that seems mostly okay?